src/script/squirrel.cpp

```diff
@@ -478,7 +478,11 @@ void Squirrel::Initialize()
 
 	/* Handle compile-errors ourself, so we can display it nicely */
 	sq_setcompilererrorhandler(this->vm, &Squirrel::CompileError);
-	sq_notifyallexceptions(this->vm, SQTrue);
+
+	SQBool notify_all = SQFalse;
+	if (_debug_script_level > 2) notify_all = SQTrue;
+		sq_notifyallexceptions(this->vm, notify_all);
```

Comment:
some odd indentation going on here. Besides, perhaps:
```suggestion
	sq_notifyallexceptions(this->vm, _debug_script_level > 2);
```

?